include/ncengine/physics/RigidBody.h

```diff
@@ -128,18 +124,19 @@ class RigidBody
         RigidBody& operator=(RigidBody&) = delete;
 
         auto GetEntity() const -> Entity { return m_self; }
-        auto GetShape() const -> Shape { return m_shape; }
+        auto GetShape() const -> const Shape& { return m_shape; }
         auto ScalesWithTransform() const -> bool { return m_flags & RigidBodyFlags::ScaleWithTransform; }
         auto GetBodyType() const -> BodyType { return m_bodyType; }
         auto GetHandle() const -> BodyHandle { return m_handle; }
+        auto IsAwake() const -> bool;
 
         void AddImpulse(const Vector3& impulse);
         void AddTorque(const Vector3& torque);
 
         auto IsInitialized() const noexcept -> bool { return m_handle; }
-        void SetPosition(const Vector3& position, bool wake = true);
-        void SetRotation(const Quaternion& rotation, bool wake = true);
-        auto SetScale(const Vector3& scale, bool wake = true) -> Vector3;
+        void SetBodyPosition(const Vector3& position, bool wake = true);
+        void SetBodyRotation(const Quaternion& rotation, bool wake = true);
+        auto SetBodyScale(const Vector3& previousScale, const Vector3& newScale, bool wake = true) -> Vector3;
```

Comment:
I wasn't quite sure. `SetBodyPosition()` and `SetBodyRotation()` don't need to be called from anything except the simulated body functions, but `SetBodyScale()` is required by client code when `ScaleWithTransform` is disabled. I left them all here, because I don't love friend functions, but I've been adding a few here and there. Still unsure.

Reply:
Oh, I think I could make the simulated body functions members of `RigidBody` without too much hassle.